app/src/main/java/com/h/pixeldroid/fragments/CameraFragment.kt

```diff
@@ -1,39 +1,283 @@
+    private lateinit var   cameraDevice: CameraDevice
+    private enum class Cameras {
+        FRONT, BACK
+    }
+
+    /* Entities necessary to the stream : */
+    private lateinit var textureView: TextureView
+
+    /* Entities necessary to the photo capture */
+    private lateinit var pictureRequestBuilder: CaptureRequest.Builder
+    private val PICK_IMAGE_REQUEST = 1
+    private val TAG = "Camera Fragment"
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
+
+        /* Views */
+
         val view = inflater.inflate(R.layout.fragment_camera, container, false)
-        val uploadPictureButton: Button = view.findViewById(R.id.upload_picture_button)
-        uploadedPictureView = view.findViewById(R.id.uploaded_picture_view)
+
+        // Get the textureView which displays the camera's stream and set up its listener :
+        // it allows us to be sure the textureView is correctly set up before performing any action
+        textureView = view.findViewById(R.id.textureView)!!
+        textureView.surfaceTextureListener = this.textureListener
+
+
+        /* Buttons */
+
+        val uploadPictureButton: Button = view.findViewById(R.id.uploadPictureButton)
         uploadPictureButton.setOnClickListener{
             uploadPicture()
         }
 
-        // Inflate the layout for this fragment
-        return view
+        val takePictureButton: Button = view.findViewById(R.id.takePictureButton)
+        takePictureButton.setOnClickListener{
+                takePicture()
+        }
+
+        return view // and start the stream building flow
+    }
+
+    private val textureListener = object : TextureView.SurfaceTextureListener {
+        override fun onSurfaceTextureAvailable(
+            surface: SurfaceTexture,
+            width: Int,
+            height: Int
+        ) {
+            openCamera(Cameras.FRONT)
+        }
+
+        override fun onSurfaceTextureSizeChanged(
+            surface: SurfaceTexture,
+            width: Int,
+            height: Int
+        ) {}
+
+        override fun onSurfaceTextureDestroyed(surface: SurfaceTexture): Boolean {
+            return false
+        }
+
+        override fun onSurfaceTextureUpdated(surface: SurfaceTexture) {}
+    }
+
+
+    private fun openCamera(cameraId: Cameras) {
+
+        manager = requireContext().applicationContext.getSystemService(Context.CAMERA_SERVICE) as CameraManager
+
+
+        /* Ask for Camera and Write permissions and require them if not granted */
+        if (ActivityCompat.checkSelfPermission(requireContext().applicationContext, Manifest.permission.CAMERA)
+            != PackageManager.PERMISSION_GRANTED)
+        {
+
+            ActivityCompat.requestPermissions(requireActivity(), arrayOf(
+                Manifest.permission.CAMERA,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE
+            ), 200)
+            openCamera(cameraId)
```

Comment:
You call recursively to not check if the user gives the permission or not when asked, right?

Reply:
My thought here is get rid of the following problem : 
1) the user is asked for the permission ; he accepts
2) he ends up with a blank screen because the function return ; he must swipe out and in of the fragment to get the desired stream
(it happens only once, but still)

However I think this permission request is not properly placed. This code will evolve in the following weeks.

Reply:
Okay I see, we'll see then :)